src/state/assets/userAssets.ts

```diff
@@ -18,18 +24,39 @@ function getOrCreateStore(address?: Address | string): ReturnType<typeof createU
   if (cachedStore && cachedAddress === accountAddress) return cachedStore;
 
   const newStore = createUserAssetsStore(accountAddress);
+
+  if (cachedStore) rebindSubscriptions(cachedStore, newStore);
+
   userAssetsStoreManager.setState({ address: accountAddress, cachedStore: newStore });
   return newStore;
 }
 
-function useUserAssetsStoreInternal<T>(selector: (state: QueryEnabledUserAssetsState) => T): T {
+function useUserAssetsStoreInternal(): QueryEnabledUserAssetsState;
+function useUserAssetsStoreInternal<T>(selector: (state: QueryEnabledUserAssetsState) => T, equalityFn?: (a: T, b: T) => boolean): T;
+function useUserAssetsStoreInternal<T>(
+  selector?: (state: QueryEnabledUserAssetsState) => T,
+  equalityFn?: (a: T, b: T) => boolean
+): QueryEnabledUserAssetsState | T {
   const address = useSelector((state: AppState) => state.settings.accountAddress);
-  return getOrCreateStore(address)(selector);
+  const store = getOrCreateStore(address);
+  return selector ? store(selector, equalityFn) : store();
 }
 
-export const useUserAssetsStore = Object.assign(useUserAssetsStoreInternal, {
+export const useUserAssetsStore: UserAssetsStoreType & {
+  getState(address?: Address | string): QueryEnabledUserAssetsState;
```

Comment:
The types here feel heavy, shouldn't all this type information be already there? I have a `withStaticProperties` helper I use in Tamagui often:

https://github.com/tamagui/tamagui/blob/main/code/core/helpers/src/withStaticProperties.tsx

Would something like that work?

Reply:
Reading a bit more, it seems you are adding these functions, I guess to me something feels off in this entire file in that there's a lot going on. Sort of forwarding the helper methods up a level, but having to re-type everything, just feels a bit much.